### ISSUE TITLE
Python 2 Unicode escapes should be parsed in search pattern

### ISSUE THREAD

**Issue by facelessuser** [facelessuser/backrefs#53]
In Python2, ur'\u0070', would be created as ur'p' when the string was instantiated.  In Python3, r'\u0070' is still `\\u0070`.  Because of this, Python3's Re library handles escaped Unicode notation, while Python2 does not.

The reason why this is a problem is that it leads to some inconsistencies. In Python3, you can construct a non-raw string like this: `'\\u0070'` and it will be treated just like `r'\u0070'`.  These are not equivalent in Python2. In backrefs, we parse case both literal unicode and `r'\u0070'` to apply casing.  This partly because this is how regex does things, but also because it makes sense as a user usually is using raw strings for replace patterns to avoid specifiying groups as `\\2` instead of the nicer `\2`, and may find it easier to use Unicode notation as well then trying to input literal Unicode.  Now on the surface this all behaves the same across all Python, but not when you use non-raw strings.

This mainly becomes a problem when you are manually constructing a string, you may use non raw string to create a string `'\\u0070'`, but while Python3 Re will parse this as Unicode, Python2 will not unless you are manually converting these.

In backrefs, we specifically handle `'\\u0070'` to apply casing and create consistency between Regex and Re, but since we don't case in search strings, we don't actually scan for non raw Unicode representations as we expected the regular expression engine to handle this, which it does, but only on Python3+.

So basically, we just want to handle scan for the `'\\u0070'` in non-raw search patterns as well.  This will keep behavior of `'\\Q\\u0070\\E'` consistent across Python versions.  Though `ur'\Q\u0070\E'` behavior will still be different for Python2 and will yield `'p'` instead of of `'\\\\u0070'`, but there is really nothing we can do about this. We can at least get behavior as close to the same as possible.

This is mainly so we can ensure that the Rummage project can send in a string from the GUI which will essentially be a non-raw string without having to reprocess the pattern looking for Unicode notation. It will still need to preprocess patterns intended for Re directly to ensure general Unicode is treated as Unicode.
